### PR TITLE
Feat/secured assets

### PR DESCRIPTION
**Pull request by Surendhar1997**
Completed working on secured resources with a time limit.

### CODE REVIEW

File: files/src/main/java/com/fincity/saas/files/dao/FilesSecuredAccessKeyDao.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		return Mono.from(
+
+		        this.dslContext.selectFrom(FILES_SECURED_ACCESS_KEY)
+		                .where(FILES_SECURED_ACCESS_KEY.ACCESS_KEY.eq(key))
+		                .limit(1))
+		        .map(e -> e.into(FilesSecuredAccessKey.class));
+
+	}
+
+	public Mono<Boolean> setAccessCount(ULong id, ULong count) {
```
Comment (serialcoder13) [inline files/src/main/java/com/fincity/saas/files/dao/FilesSecuredAccessKeyDao.java:32]: It should be an increasing count, and the query has to be counted + 1 in the update. and the condition has to be where id = given id and accessed_count < limit. That way you can guarantee the concurrency.

File: files/src/main/java/com/fincity/saas/files/service/FilesAccessPathService.java [hunk 2]
```diff
@@ -303,7 +303,8 @@ public Mono<Integer> delete(ULong id) {
 	public Mono<Boolean> hasReadAccess(String actualPath, String clientCode, FilesAccessPathResourceType resourceType) {
 
 		String path = actualPath.endsWith("/") ? actualPath.substring(0, actualPath.length() - 1) : actualPath;
-		Mono<String> mKey = cacheService.makeKey(clientCode, ":", resourceType, ":read:", path);
+
+		Mono<String> mKey = Mono.just(clientCode + ":" + resourceType + ":read:" + path);
 
 		return mKey.flatMap(e -> cacheService.<Boolean>get(CACHE_NAME_ACCESS_PATH, e))
```
Comment (serialcoder13) [inline files/src/main/java/com/fincity/saas/files/service/FilesAccessPathService.java:309]: We need to update the way we are storing the values in the Cache. Use cacheOrGet method in the cache services.

File: files/src/main/java/com/fincity/saas/files/service/FilesSecuredAccessService.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	public Mono<String> getAccessPathByKey(String accessKey) {
+
+		return FlatMapUtil.flatMapMono(
+
+		        () -> this.getAccessRecordByPath(accessKey),
+
+		        accessKeyObject -> this.checkAccessWithinTime(accessKeyObject)
+		                .flatMap(BooleanUtil::safeValueOfWithEmpty),
```
Comment (serialcoder13) [inline files/src/main/java/com/fincity/saas/files/service/FilesSecuredAccessService.java:33]: Why do you need the SafeValueOfWithEmpty??? You will be returning either true or false in this case right.

**Review by serialcoder13 — COMMENTED** [review]
Please update according to the comments.

File: files/src/main/java/com/fincity/saas/files/dao/FilesSecuredAccessKeyDao.java (hunk 1 above)
Comment (Surendhar1997) [reply files/src/main/java/com/fincity/saas/files/dao/FilesSecuredAccessKeyDao.java:32]: Updated the query with mentioned comment.

File: configfiles/application-default.yml [hunk 4]
```diff
@@ -17,6 +17,8 @@ files:
     location:
       static: C:/fincity/resources/static
       secured: C:/fincity/resources/secured
+  secureKeyURI: /api/files/secured/downloadFileByKey/
```
Comment (serialcoder13) [inline configfiles/application-default.yml:20]: Remove this and default it like I showed you yesterday, it is not going to change easily

**Review by serialcoder13 — COMMENTED** [review]
Please change that one change...

File: configfiles/application-default.yml (hunk 4 above)
Comment (serialcoder13) [reply configfiles/application-default.yml:20]: Also it should not start with /

Comment (Surendhar1997) [reply configfiles/application-default.yml]: done.

**Review by serialcoder13 — APPROVED** [review]
Looks good

**Pull request merged**